test/test-remote/looker/src/index.ts

```diff
@@ -177,10 +177,25 @@ async function createNewLogSeriess(
 
     let stream: MetricSeries | LogSeries;
     if (CFG.metrics_mode) {
+      // Calculate amount to subtract from start time, or avoid random() if subtraction is disabled
+      const subtractSecs = (CFG.metrics_past_start_range_max_seconds === 0)
+        ? 0
+        // By default this is a time between now-30min and now-20min - smear
+        // this out by plus/minus 5 minutes, because of the throttling
+        // mechanism otherwise hitting in for all series at the same time.
+        // Note that Math.random() returns [0,1) (not including 1).
+        : util.rndFloatFromInterval(
+            CFG.metrics_past_start_range_min_seconds,
+            CFG.metrics_past_start_range_max_seconds
+        );
       stream = new MetricSeries(
         {
-          metricName: `looker_${rndstring(4)}`, // might collide among streams, which is OK as long as the label set adds uniqueness
-          uniqueName: streamname, // must not collide among streams
+          // the same across concurrent streams, contains invocation_id
+          // ensure any dashes in metric name are switched to underscores: required by prometheus
+          // see also: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+          metricName: guniqueCycleId.replace(/-/g, "_"),
```

Comment:
The idea is that `--n-concurrent-streams` can be O(10^6). That is, we cannot use the same metric name for all time series written in a cycle.

I think for simplicity we should have each metric name be unique.

If we want to be a little more sophisticated we can use the same metric name for M time series, with M guaranteed to be below an upper bound. (but before building _that_, I thought, it makes sense to simply have unique metric names).

Which requirements do we have for the metric name, maybe in terms of predictability and discoverability? I did use the looker log for finding metric names, and used those to go into Grafana Explore for querying.

Reply:
Oh good point. I'll switch this back to being unique - as of this commit it's sharing the same metric name across the concurrent streams, with the difference being in the uniquename label. This also would explain the CI failure with hitting the 50k series cap.

At some point it'd be useful to be able to configure a number of dynamic labels to be included in the metrics to simulate a high-cardinality metric. So for example a test scenario could be to have the metrics each with 10 dynamic labels, with 1000 distinct label permutations against the same metric name. having `metricName=guniqueCycleId` sort of did this - where the number of distinct labels for a common metric name were based on the concurrency value. But this is _really_ unintuitive, and it'd be better to just have an explicit option for label cardinality.

For the names themselves I've personally just been testing things write-only so far and hadn't been thinking too much about the metric names themselves outside of configuring the cardinality.

Reply:
Btw, let's think of `--n-concurrent-streams` more like `--n-series-per-cycle` -- the "concurrent" part became more or less meaningless / misleading when I introduced the `--max-concurrent-[writes|reads]`

Reply:
>  But this is really unintuitive, and it'd be better to just have an explicit option for label cardinality.

We could think of something like `--reuse-metric-name-across-n-series N`. Then require `--n-concurrent-streams N` to be larger than that. Then automatically add further labels to achieve this.

With "cardinatlity" AFAIK being defined as the number of unique combinations we could instead call this thing `--metrics-cardinality C`. We might want to default that to 1 or to 10.

> At some point it'd be useful to be able to configure a number of dynamic labels to be included in the metrics to simulate a high-cardinality metric

With the above's approach, labels would be added automatically. However, that of course may require guidance. I would say, the parameter that might be most useful here is the maximum number of values for a specific label, e.g. `--max-values-per-label M`. We might default that to 10. 

After all, this does not require setting the "number of dynamic labels"; that would be one degree of freedom too much.